### PR TITLE
Add `semanticLabel` to `CupertinoCheckbox`

### PR DESCRIPTION
**Pull request by victorsanni**
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by MitchellGoodwin — COMMENTED** [review]
The adaptive checkbox should also make use of this new property in this PR.

File: packages/flutter/lib/src/cupertino/checkbox.dart [hunk 1]
```diff
@@ -165,6 +166,13 @@ class CupertinoCheckbox extends StatefulWidget {
   /// [RoundedRectangleBorder] with a circular corner radius of 4.0.
   final OutlinedBorder? shape;
 
+  /// The semantic label for the checkbox that will be announced by screen readers.
+  ///
+  /// This is announced in accessibility modes (e.g TalkBack/VoiceOver).
```
Comment (chunhtai) [inline packages/flutter/lib/src/cupertino/checkbox.dart:171]:
```suggestion
  /// This is announced by assistive technologies (e.g TalkBack/VoiceOver).
```

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/cupertino/checkbox.dart (hunk 1 above)
Comment (MitchellGoodwin) [reply packages/flutter/lib/src/cupertino/checkbox.dart:171]: @chunhtai looks like this came from the description from the same property in the Material checkbox:
https://api.flutter.dev/flutter/material/Checkbox/semanticLabel.html

Should your suggested wording be used in both?

**Review by MitchellGoodwin — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/cupertino/checkbox.dart (hunk 1 above)
Comment (victorsanni) [reply packages/flutter/lib/src/cupertino/checkbox.dart]: I changed the wording for other semanticLabel API docs across the framework.

Comment (chunhtai) [reply packages/flutter/lib/src/cupertino/checkbox.dart]: yes, assistive technologies are what we use to address voiceover or talkback. We decided to use this word about 2 to 3 year ago, so there may be some old wording in the code base. I would update them all if found any.

**Pull request merged**